flytekit/core/interface.py

```diff
@@ -70,6 +71,8 @@ def __init__(
         self._inputs: Union[Dict[str, Tuple[Type, Any]], Dict[str, Type]] = {}  # type: ignore
         if inputs:
             for k, v in inputs.items():
+                if not k.isidentifier():
+                    raise ValueError(f'Input must be valid Python identifier: {k!r}')
```

Comment:
Adds validation on input naming in `Interface`. Note that this is already done on the outputs:
```
variables = [k for k in outputs.keys()]
...
collections.namedtuple(..., variables) # collections.namedtuple checks the variables are valid python identifiers
```
So it is sensible to add similar validation to the inputs, which must also by valid python types.